### PR TITLE
Implemented initial version of the Refactor Yaml Parameter Reading

### PR DESCRIPTION
**Pull request by alexanderKoltman**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -99,6 +99,32 @@ You can have the tool create a machine-readable log file by specifying the `--lo
   -y, --yaml=<yamlFile>                  The path to the yaml file. (Not needed for the get commands)
 ```
 
+##### [HINT - DEFAULT VALUES FOR SOME PARAMS]
```
Comment (M4eqP) [inline README.md:102]: I wouldn't call these default values, that's a bit misleading. Maybe adjust the title to point out it's about data in the YAML configs.

File: cloud.foundry.cli/src/main/java/cloud/foundry/cli/crosscutting/mapping/CfOperationsCreator.java [hunk 2]
```diff
@@ -42,11 +56,49 @@ public static DefaultCloudFoundryOperations createCfOperations(LoginCommandOptio
...
                 .uaaClient(reactorUaaClient)
-                .organization(commandOptions.getOrganization())
-                .space(commandOptions.getSpace())
+                .organization(targetBean.getOrg())
+                .space(targetBean.getSpace())
                 .build();
     }
 
+    private static TargetBean replaceTargetOptions(TargetBean targetBean, LoginCommandOptions commandOptions) {
+        if (targetBean == null) targetBean = new TargetBean();
```
Comment (M4eqP) [inline cloud.foundry.cli/src/main/java/cloud/foundry/cli/crosscutting/mapping/CfOperationsCreator.java:65]: Please move into a scope.

File: cloud.foundry.cli/src/main/java/cloud/foundry/cli/crosscutting/mapping/CfOperationsCreator.java [hunk 3]
```diff
@@ -25,14 +32,21 @@
      * {@link cloud.foundry.cli.operations}. The cfOperations object
      *
      * @param commandOptions {@link LoginCommandOptions}
+     * @param targetBean     Bean that holds all data that is related to the target space.
```
Comment (felix-oq) [inline cloud.foundry.cli/src/main/java/cloud/foundry/cli/crosscutting/mapping/CfOperationsCreator.java:35]: Maybe explicitly allow null values here:
```suggestion
     * @param targetBean     Bean that holds all data that is related to the target space (can be null)
```

File: cloud.foundry.cli/src/main/java/cloud/foundry/cli/logic/GetLogic.java [hunk 4]
```diff
@@ -36,7 +36,7 @@
      * @param servicesOperations ServicesOperations
      * @param applicationsOperations ApplicationsOperations
      * @param clientOperations ClientOperations
-     * @param loginOptions LoginCommandOptions
+     * @param loginOptions ILoginCommandOptions
```
Comment (felix-oq) [inline cloud.foundry.cli/src/main/java/cloud/foundry/cli/logic/GetLogic.java:39]: Should be renamed back to `LoginCommandOptions`

File: cloud.foundry.cli/src/main/java/cloud/foundry/cli/services/LoginCommandOptions.java [hunk 5]
```diff
@@ -1,73 +1,14 @@
 package cloud.foundry.cli.services;
 
-import picocli.CommandLine.Option;
-import picocli.CommandLine.ScopeType;
+public interface LoginCommandOptions {
```
Comment (felix-oq) [inline cloud.foundry.cli/src/main/java/cloud/foundry/cli/services/LoginCommandOptions.java:3]: Documentation should be added in this interface

File: cloud.foundry.cli/src/main/java/cloud/foundry/cli/services/OptionalLoginCommandOptions.java [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+
+import picocli.CommandLine.Option;
+import picocli.CommandLine.ScopeType;
+
+/**
+ * Common options for the initialization process of the
+ * {@link org.cloudfoundry.operations.CloudFoundryOperations operations} object.
+ *
+ * @see cloud.foundry.cli.crosscutting.mapping.CfOperationsCreator
+ */
```
Comment (felix-oq) [inline cloud.foundry.cli/src/main/java/cloud/foundry/cli/services/OptionalLoginCommandOptions.java:11]: The documentation should mention that the user is not forced to provide target information for these `LoginCommandOptions`

File: cloud.foundry.cli/src/main/java/cloud/foundry/cli/services/RequiredLoginCommandOptions.java [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+import static picocli.CommandLine.Option;
+
+import picocli.CommandLine;
+
+/**
+ * Options for the initialization process of the
+ * {@link org.cloudfoundry.operations.CloudFoundryOperations operations} object.
+ *
+ * @see cloud.foundry.cli.crosscutting.mapping.CfOperationsCreator
+ */
```
Comment (felix-oq) [inline cloud.foundry.cli/src/main/java/cloud/foundry/cli/services/RequiredLoginCommandOptions.java:12]: The documentation should mention that the user is forced to provide target information for these `LoginCommandOptions`

File: cloud.foundry.cli/src/main/java/cloud/foundry/cli/crosscutting/mapping/CfOperationsCreator.java (hunk 3 above)
Comment (M4eqP) [reply cloud.foundry.cli/src/main/java/cloud/foundry/cli/crosscutting/mapping/CfOperationsCreator.java:35]: That's a good idea. For commands which don't accept YAML files, one doesn't need to pass any a bean at all then.

File: cloud.foundry.cli/src/main/java/cloud/foundry/cli/logic/GetLogic.java (hunk 4 above)
Comment (M4eqP) [reply cloud.foundry.cli/src/main/java/cloud/foundry/cli/logic/GetLogic.java:39]: Repeating the class name in the param description doesn't make a lot of sense by the way. Please consider adding an actual description of the parameter.

File: cloud.foundry.cli/src/main/java/cloud/foundry/cli/logic/GetLogic.java [hunk 8]
```diff
@@ -36,7 +36,7 @@
      * @param servicesOperations ServicesOperations
      * @param applicationsOperations ApplicationsOperations
      * @param clientOperations ClientOperations
-     * @param loginOptions ILoginCommandOptions
+     * @param loginOptions user provided Login-Options
```
Comment (felix-oq) [inline cloud.foundry.cli/src/main/java/cloud/foundry/cli/logic/GetLogic.java]: The other parameters still don't have an explaination

File: README.md [hunk 9]
```diff
@@ -99,7 +99,7 @@ You can have the tool create a machine-readable log file by specifying the `--lo
   -y, --yaml=<yamlFile>                  The path to the yaml file. (Not needed for the get commands)
 ```
 
-##### [HINT - DEFAULT VALUES FOR SOME PARAMS]
+##### [HINT - TARGET BEAN INFORMATION FOR SOME PARAMS]
```
Comment (felix-oq) [inline README.md:103]:
```suggestion
##### [HINT - TARGET INFORMATION FOR SOME COMMANDS]
```

File: README.md [hunk 10]
```diff
@@ -99,7 +99,7 @@ You can have the tool create a machine-readable log file by specifying the `--lo
...
 
 For the <code>diff/apply</code> commands, the parameters can be fetched from the target section of the given YAML file,
 to reduce the number of the program target information `api`, `organization` and `space`.
```
Comment (felix-oq) [inline README.md]: The sentence didn't quite make sence:
```suggestion
For the `diff/apply` commands, some parameters can be fetched from the target section of the given YAML file.
```

**Pull request merged**